File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        for(int i = 0; i < nums.size(); ++i)
        {
            for (int j = i+1; j < nums.size(); ++j)
            {
                if (i == j) return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ret = []
        nums = sorted(nums)

        for i, n in enumerate(nums):
            start = 0
            end = len(nums)-1

            while start < end:
                if start == i: start += 1 
                elif end == i: end -= 1 
                elif nums[start] + nums[end] > -n: end-=1 
                elif nums[start] + nums[end] < -n: start+=1 
                else: 
                    temp = sorted([n, nums[start], nums[end]])
                    if temp not in ret: ret.append(temp)
                    break

        return ret